oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java

```diff
@@ -950,14 +985,19 @@ public void getRequestMetadata(
   /** Provide the request metadata by putting an access JWT directly in the metadata. */
   @Override
   public Map<String, List<String>> getRequestMetadata(URI uri) throws IOException {
-    if (scopes.isEmpty() && defaultScopes.isEmpty() && uri == null) {
-      throw new IOException(
-          "Scopes and uri are not configured for service account. Either pass uri"
-              + " to getRequestMetadata to use self signed JWT, or specify the scopes"
-              + " by calling createScoped or passing scopes to constructor.");
+    if (createScopedRequired()) {
```

Comment:
i think we don't want to use this property here. It seems like not correct at the moment, because CreateScoped might not be actually required and Scopes could be empty at the same time. You can see functional tests examples as the reference. First couple 200 test cases will return true for createScopedRequired. The reason is that, as I understand, ESF will set scope based on audience. 

Therefore, I think that property is subject to change in the future.

Reply:
This is the original behavior of the library. I would like to keep the original behavior in this PR so it doesn't cause any problems.

Reply:
yep you are right, either scopes or uri work.

Reply:
Nope. I would like to revert to the commit before phrase 1. In other words, always use scopes and never use uri.

https://github.com/googleapis/google-auth-library-java/blob/497d4e739cf83876b77d1eda8c9618968cf8165a/oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java#L414-L418

The integration test is not correct, it should be updated. It works because it uses self signed jwt with a modified url as audience (used the `https://<api>.googleapis.com/` part from the url). This is what I introduced in phrase 1. However this logic is wrong, url may be a regional endpoint or any url unrelated to audience, there is no way to derive the audience from uri. This is also the reason why I reverted the url modification logic in ServiceAccountJwtAccessCredentials in this PR. 
 
In other words, we should always scopes (never use uri) for service account credentials for Java. In this PR if `useJwtAccessWithScope` is `true`, I still included the `uri` logic because it won't be used by client libraries - GAPIC clients always set scopes we will always use scopes.

Reply:
I updated the PR to only use scopes and never use uri, and updated the functional tests.

Reply:
Thanks for explaining the change! However, the fact that those functional tests will fail with the change - means this is a breaking change, right? People could have used it this way already.

Reply:
added uri back